test/linalg/qr.jl

```diff
@@ -20,150 +19,152 @@ a2img  = randn(n,n)/2
+                    if eltya != Int
+                        @test eye(eltyb,n)*q ≈ convert(AbstractMatrix{tab},q)
+                    end
                 end
             end
         end
 
-debug && println("Matmul with QR factorizations")
         if eltya != Int
-            qrpa = factorize(a[:,1:n1])
-            q, r = qrpa[:Q], qrpa[:R]
-            @test A_mul_B!(full(q, thin=false)',q) ≈ eye(n)
-            @test_throws DimensionMismatch A_mul_B!(eye(eltya,n+1),q)
-            @test A_mul_Bc!(full(q, thin=false),q) ≈ eye(n)
-            @test_throws DimensionMismatch A_mul_Bc!(eye(eltya,n+1),q)
-            @test_throws BoundsError size(q,-1)
-            @test_throws DimensionMismatch Base.LinAlg.A_mul_B!(q,zeros(eltya,n1+1))
-            @test_throws DimensionMismatch Base.LinAlg.Ac_mul_B!(q,zeros(eltya,n1+1))
-
-            qra = qrfact(a[:,1:n1], Val(false))
-            q, r = qra[:Q], qra[:R]
-            @test A_mul_B!(full(q, thin=false)',q) ≈ eye(n)
-            @test_throws DimensionMismatch A_mul_B!(eye(eltya,n+1),q)
-            @test A_mul_Bc!(full(q, thin=false),q) ≈ eye(n)
-            @test_throws DimensionMismatch A_mul_Bc!(eye(eltya,n+1),q)
-            @test_throws BoundsError size(q,-1)
-            @test_throws DimensionMismatch q * eye(Int8,n+4)
+            @testset "Matmul with QR factorizations" begin
+                qrpa = factorize(a[:,1:n1])
+                q, r = qrpa[:Q], qrpa[:R]
+                @test A_mul_B!(full(q, thin=false)',q) ≈ eye(n)
+                @test_throws DimensionMismatch A_mul_B!(eye(eltya,n+1),q)
+                @test A_mul_Bc!(full(q, thin=false),q) ≈ eye(n)
+                @test_throws DimensionMismatch A_mul_Bc!(eye(eltya,n+1),q)
+                @test_throws BoundsError size(q,-1)
+                @test_throws DimensionMismatch Base.LinAlg.A_mul_B!(q,zeros(eltya,n1+1))
+                @test_throws DimensionMismatch Base.LinAlg.Ac_mul_B!(q,zeros(eltya,n1+1))
+
+                qra = qrfact(a[:,1:n1], Val(false))
+                q, r = qra[:Q], qra[:R]
+                @test A_mul_B!(full(q, thin=false)',q) ≈ eye(n)
+                @test_throws DimensionMismatch A_mul_B!(eye(eltya,n+1),q)
+                @test A_mul_Bc!(full(q, thin=false),q) ≈ eye(n)
+                @test_throws DimensionMismatch A_mul_Bc!(eye(eltya,n+1),q)
+                @test_throws BoundsError size(q,-1)
+                @test_throws DimensionMismatch q * eye(Int8,n+4)
+            end
         end
     end
 end
 
-# Because transpose(x) == x
-@test_throws ErrorException transpose(qrfact(randn(3,3)))
-@test_throws ErrorException ctranspose(qrfact(randn(3,3)))
-@test_throws ErrorException transpose(qrfact(randn(3,3), Val(false)))
-@test_throws ErrorException ctranspose(qrfact(randn(3,3), Val(false)))
-@test_throws ErrorException transpose(qrfact(big.(randn(3,3))))
-@test_throws ErrorException ctranspose(qrfact(big.(randn(3,3))))
+@testset "transpose errors" begin
+    # Because transpose(x) == x
```

Comment:
Yes. We don't have the fallback anymore so the comment is outdated. We can probably also remove the methods in `factorizations.jl` that used to be exercised here and change the error tests to `MethodError`.